entity_emailer/interface.py

```diff
@@ -90,16 +93,18 @@ def send_unsent_scheduled_emails(cls):
                 cls.save_email_exception(email, traceback.format_exc())
 
         # Send all the emails that were generated properly
+        emails_sent_ids = []
         with mail.get_connection() as connection:
             for email in emails_to_send:
                 try:
                     # Send mail
                     connection.send_messages([email.get('message')])
+                    emails_sent_ids.append(email.get('model').id)
                 except Exception as e:
                     cls.save_email_exception(email.get('model'), e)
 
         # Update the emails as sent
-        to_send.update(sent=current_time)
+        Email.objects.filter(id__in=emails_sent_ids).update(sent=current_time)
```

Comment:
I think in this scenario we are going to have to save each one of these one at a time inside of the loop. Otherwise you run into a similar situation where if this process gets terminated in the middle of sending 1000 emails and you are on number 900, the next time the process starts you will be sending 900 duplicate emails. Unfortunately its inefficient but I cant think of another alternative that fully solves the problem